File: AtCoder_Beginner_Contest_126/B_YYMM_or_MMYY.js
var lines = [];
var readline = require('readline');

var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

rl.on('line', function (x) {
    lines.push(x);
});

rl.on('close', function () {
    var S = lines[0].split("").map(i => Number(i));
    var a = S[0] * 10 + S[1]
    var b = S[2] * 10 + S[3]

    if (a <= 12 && a !== 0) {
        if (b <= 12 && b !== 0) {
            console.log("AMBIGUOUS"); return (0)
        }
        console.log("MMYY"); return (0);
    }

    if (a > 12 || a === 0) {
        if (b <= 12 && b !== 0) {
            console.log("YYMM"); return (0);
        }
        console.log("NA"); return (0);
    }
    console.log("NA")
});

File: AtCoder_Beginner_Contest_126/B_YYMM_or_MMYY.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var S string
	fmt.Scan(&S)
	s := strings.Split(S, "")

	w, _ := strconv.Atoi(s[0])
	x, _ := strconv.Atoi(s[1])
	y, _ := strconv.Atoi(s[2])
	z, _ := strconv.Atoi(s[3])

	// MM
	if 0 < w*10+x && w*10+x <= 12 {
		if 0 < y*10+z && y*10+z <= 12 {
			fmt.Println("AMBIGUOUS")
		} else {
			fmt.Println("MMYY")
		}
	} else {
		if 0 < y*10+z && y*10+z <= 12 {
			fmt.Println("YYMM")
		} else {
			fmt.Println("NA")
		}
	}
}
